src/main/java/uk/gov/pay/api/utils/AuthorisationSummaryHelper.java

```diff
@@ -0,0 +1,12 @@
+package uk.gov.pay.api.utils;
+
+import uk.gov.pay.api.model.AuthorisationSummary;
+
+public class AuthorisationSummaryHelper {
+    public static AuthorisationSummary includeAuthorisationSummaryWhen3dsRequired(AuthorisationSummary authorisationSummary) {
+        if (authorisationSummary == null || authorisationSummary.getThreeDSecure() == null || !authorisationSummary.getThreeDSecure().isRequired()) {
+            return null;
+        }
+        return authorisationSummary;
+    }
```

Comment:
This might be a bit clearer if it’s inverted:

```java
if (authorisationSummary != null && authorisationSummary.getThreeDSecure() != null && authorisationSummary.getThreeDSecure().isRequired()) {
    return authorisationSummary;
}
return null;
```

But I think the difference is fairly marginal so I’m not going to insist on it.

Reply:
Inverted